lib/IRGen/Linking.cpp

```diff
@@ -376,8 +376,10 @@ std::string LinkEntity::mangleAsString() const {
   }
 
   case Kind::CoroutineContinuationPrototype:
-    return mangler.mangleCoroutineContinuationPrototype(
-                                            cast<SILFunctionType>(getType()));
+    // Emit unnamed functions for coroutine continuation
```

Comment:
The mangler is unable to handle primary archetypes (as shown in one of the testcases). However, mangling is unnecessary here as coroutine prototype name is not ABI-specific, the coroutine lowering pass in LLVM splits and renames the function anyway, we only need the name to be unique.

Reply:
Okay. 

If we want to get the ability back we can add the following code and reinstate the mangling:

```
--- a/lib/IRGen/GenFunc.cpp
+++ b/lib/IRGen/GenFunc.cpp
@@ -1456,8 +1456,11 @@ public:
         Self(nullptr), FirstData(nullptr), SecondData(nullptr) {}
 
   void begin() override {
+    auto unsubstType = substType->getUnsubstitutedType(IGM.getSILModule());
     auto prototype = subIGF.IGM.getOpaquePtr(
-        subIGF.IGM.getAddrOfContinuationPrototype(origType));
+        subIGF.IGM.getAddrOfContinuationPrototype(
+          cast<SILFunctionType>(
+            unsubstType->mapTypeOutOfContext()->getCanonicalType())));
 
     // Use malloc and free as our allocator.
     auto allocFn = subIGF.IGM.getOpaquePtr(subIGF.IGM.getMallocFn());
```